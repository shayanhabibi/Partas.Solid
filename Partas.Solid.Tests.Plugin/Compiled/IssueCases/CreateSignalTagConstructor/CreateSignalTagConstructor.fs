module Partas.Solid.Tests.IssueCases.CreateSignalTagConstructor

// #2
// createSignal was being converted into <createSignal /> due to a nested tagConstructor
// combined with the fact that it was a user import at the top level. Or something along those lines.
// Fixed by making the tag constructors match for ".ctor" in the member info

open Partas.Solid
open Fable.Core
open CreateSignalTagConstructor.Types

[<JS.Pojo>]
type MenuItem(title: string, url: string, icon: RegularNode) =
    member val title = title
    member val url = url
    member val icon = icon
module App =
    [<Import("fakeTag", "fakeLibrary")>]
    type [<Erase>] MyItemizer() =
        interface RegularNode
        member val items: MenuItem array = unbox null with get,set
        [<SolidTypeComponent>]
        member props.constructor =
            div()
    
    [<SolidComponent>]
    let App() =
        let initialItems = [|
            MenuItem("Home", "#", div())
        |]
        let (items, setItems) = createSignal(initialItems)
        let (item, addItem) = createSignal<MenuItem>(Unchecked.defaultof<_>)
        div() {
            MyItemizer(items = items())
            Imported()
            Match()
        }
