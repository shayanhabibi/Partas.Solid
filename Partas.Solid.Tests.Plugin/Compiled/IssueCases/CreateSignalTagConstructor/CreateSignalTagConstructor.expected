import { Match, createSignal } from "solid-js";
import { defaultOf } from "../../fable_modules/fable-library-js.5.0.0-alpha.13/Util.js";
import { fakeTag } from "fakeLibrary";
import { FakeImport } from "FakeLibrary";

export function App_App() {
    const patternInput = createSignal([{
        title: "Home",
        url: "#",
        icon: <div />,
    }]);
    const patternInput_1 = createSignal(defaultOf());
    return <div>
        <fakeTag items={patternInput[0]()} />
        <FakeImport />
        <Match />
    </div>;
}

